### PR TITLE
feat: gRPC prerequisites/dependencies

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-597

**Changes:**
- Prerequisites for gRPC PR.

-----------

### CODE REVIEW

File: internal/pkg/service/stream/storage/level/local/model/file.go [hunk 1]
```diff
@@ -13,6 +12,4 @@ type File struct {
 	Assignment assignment.Assignment `json:"assignment"`
 	// Allocation configures pre-allocation of the filesystem free space.
 	Allocation diskalloc.Config `json:"allocation"`
-	// Encoding defines how is the  result format encoded, for example tabular data to CSV.
-	Encoding encoding.Config `json:"encoding"`
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/model/file.go:17]: Sorry, one more smaller refactoring, I need to pass only required configuration to the disk writer node, 

so `{file/slice}.LocalStorage.Encoding` has been moved to `{file/slice}.Encoding`, it is also related to some previous comment, in a previous PR.

Encoding is handled by the source node, in the "local" environment/level, ... but disk writer is providing "local storage",

File: internal/pkg/service/stream/keboolasink/bridge/fixtures/create_sink_snapshot.txt [hunk 2]
```diff
@@ -148,7 +148,8 @@ storage/slice/level/local/123/456/my-source/my-sink/2000-01-01T01:00:00.000Z/my-
   },
   "local": {
     "dir": "123/456/my-source/my-sink/2000-01-01T01-00-00-000Z/2000-01-01T01-00-00-000Z",
-    "filename": "slice.csv.gz",
+    "filenamePrefix": "slice",
+    "filenameExtension": "csv.gz",
     "allocatedDiskSpace": "100MB"
```
Comment (michaljurecko) [inline internal/pkg/service/stream/keboolasink/bridge/fixtures/create_sink_snapshot.txt]: I need to split the `filename` field to the two, will be explained later (writing from each source node requires a separate disk file).

File: internal/pkg/service/stream/storage/level/local/diskwriter/volume.go [hunk 3]
```diff
@@ -176,16 +177,16 @@ func (v *Volume) Metadata() volume.Metadata {
 	}
 }
 
-func (v *Volume) OpenWriter(slice *model.Slice) (w Writer, err error) {
+func (v *Volume) OpenWriter(sliceKey model.SliceKey, slice localModel.Slice) (w Writer, err error) {
 	// Check context
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/diskwriter/volume.go:181]: Optimalization for etcd mirror.

File: internal/pkg/service/stream/components.go [hunk 4]
```diff
@@ -27,26 +29,62 @@ type Components []Component
...
 	for _, c := range components {
 		componentsMap[c] = true
 	}
 
+	// Common distribution scope
+	var distScp commonDeps.DistributionScope
+	if componentsMap[ComponentStorageWriter] || componentsMap[ComponentHTTPSource] || componentsMap[ComponentStorageCoordinator] {
+		distScp = commonDeps.NewDistributionScope(cfg.NodeID, cfg.Distribution, d)
+	}
+
```
Comment (michaljurecko) [inline internal/pkg/service/stream/components.go:43]: I updated `dependencies` scopes, some of them were missing.

File: internal/pkg/service/stream/storage/level/local/model/file.go (hunk 1 above)
Comment (Matovidlo) [reply internal/pkg/service/stream/storage/level/local/model/file.go:17]: It's ok but now it makes more sense :smile:

File: internal/pkg/service/stream/components.go [hunk 5]
```diff
@@ -27,26 +29,62 @@ type Components []Component
...
 
-func StartComponents(ctx context.Context, d dependencies.ServiceScope, cfg config.Config, components ...Component) error {
+func StartComponents(ctx context.Context, d dependencies.ServiceScope, cfg config.Config, components ...Component) (err error) {
 	componentsMap := make(map[Component]bool)
 	for _, c := range components {
 		componentsMap[c] = true
 	}
 
+	// Common distribution scope
+	var distScp commonDeps.DistributionScope
```
Comment (Matovidlo) [inline internal/pkg/service/stream/components.go:39]: `distScp` -> `distScope`, Combination of `dist` and `scp` is very weird as,  Scp is command for secure copy files, and I was unsure why we use that.

**Review by Matovidlo — APPROVED** [review]
LGTM

File: internal/pkg/service/stream/components.go (hunk 5 above)
Comment (michaljurecko) [reply internal/pkg/service/stream/components.go:39]: Rename it however you want :slightly_smiling_face:  , but please in the whole repository.

**Pull request merged**